### PR TITLE
RefineTypes fixes for TOSA backend

### PR DESCRIPTION
**Pull request by sjarus**
Handles Linear, Adaptive_AvgPool2D and FlattenUsintInts
Adds ResNet18 static model for TOSA

Signed-off-by: Suraj Sudhir <suraj.sudhir@arm.com>

### CODE REVIEW

File: e2e_testing/torchscript/vision_models.py [hunk 1]
```diff
@@ -12,36 +12,67 @@
...
         torch.manual_seed(0)
         self.resnet = models.resnet18()
         self.train(False)
+
     @export
-    @annotate_args([
-        None,
-        ([-1, 3, -1, -1], torch.float32, True),
-    ])
+    @annotate_args(
```
Comment (cathyzhyi) [inline e2e_testing/torchscript/vision_models.py:25]: nit: the testing framework has been using the existing format for `annotate_args `. could you change it back for consistency?

File: e2e_testing/torchscript/vision_models.py [hunk 2]
```diff
@@ -12,36 +12,67 @@
...
+    def __init__(self):
+        super().__init__()
+        # Reset seed to make model deterministic.
+        torch.manual_seed(0)
+        self.resnet = models.resnet18()
+        self.train(False)
+
+    @export
+    @annotate_args(
+        [
```
Comment (cathyzhyi) [inline e2e_testing/torchscript/vision_models.py:50]: nit: annotate_args format.

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp [hunk 3]
```diff
@@ -1116,7 +1129,16 @@ ChangeResult TypeAnalyzer::visitAtenFlattenUsingIntsOp(
       knowledge.hasSizes = true;
       for (auto i = 0; i < startDim; i++)
         knowledge.sizes.push_back(operand.sizes[i]);
-      knowledge.sizes.push_back(kUnknownSize);
+      int64_t dimProduct = 1;
+      bool hasDynamicDims = false;
+      for (auto i = startDim; i <= endDim; i++) {
+        if (operand.sizes[i] == kUnknownSize) {
+          hasDynamicDims = true;
```
Comment (cathyzhyi) [inline lib/Dialect/Torch/Transforms/RefineTypes.cpp:1136]: nit: could let `dimProduct = kUnknownSize ` here, then `hasDynamicDims ` can be removed.

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

File: e2e_testing/torchscript/vision_models.py (hunk 1 above)
Comment (sjarus) [reply e2e_testing/torchscript/vision_models.py:25]: Yeah I'm not sure how this happened. Probably VSCode autoformatting.

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp (hunk 3 above)
Comment (sjarus) [reply lib/Dialect/Torch/Transforms/RefineTypes.cpp:1136]: Good point. I didn't look up what kUnknownSize was.

**Review by cathyzhyi — APPROVED** [review]

**Pull request merged**